pkg/proxy/proxy.go

```diff
@@ -186,10 +186,14 @@ type invocationError struct {
 
 func writeError(writer http.ResponseWriter, err error, accept string) {
 	accepts := goautoneg.ParseAccept(accept)
-	preferJSON := accepts[0].Type == "application" && accepts[0].SubType == "json"
-
-	if preferJSON {
-		writer.Header().Set("content-type", "application/json")
+	var preferJSON bool
```

Comment:
since we rely on this value further down, I'd favor explicit initialization rather than leaning on the default value

```suggestion
	preferJSON := false
```